lib/editor/actions/map/stopStrategies.js

```diff
@@ -553,3 +561,175 @@ export function removeStopFromPattern (pattern: Pattern, stop: GtfsStop, index:
     dispatch(saveActiveGtfsEntity('trippattern'))
   }
 }
+
+/**
+ * Based on the provided starting point for deletion and the number of segments until we hit another stop,
+ * this function calculates the number of control points and adds the control points and segments to the array
+ * for later deletion.
+ */
+const updateControlPointsAndSegments = (deletedControlPoints: Array<number>, deletedSegments: Array<number>, deletionStartIndex: number, numberSegments: number): Array<Array<number>> => {
+  // Each control point creates 2 segments, so you have one less control point than segments.
+  const numberControlPoints = numberSegments - 1
+  // Add control points and segments to the list to delete and return
+  return [
+    addRangeToArray(deletedControlPoints, deletionStartIndex + 1, numberControlPoints), // Add one for immediate next control point.
+    addRangeToArray(deletedSegments, deletionStartIndex, numberSegments)
+  ]
+}
+
+/**
+ * Generates a new segment for the fromPoint and toPoint and inserts it into the new patternSegments array.
+ */
+const generateSegmentAndInsert = async (clonedPatternSegments: Array<Coordinates>, fromPoint: ?ControlPoint, toPoint: ?ControlPoint, insertionPoint: number) => {
+  if (!fromPoint || !toPoint) return
+  const points = [fromPoint.point.geometry.coordinates, toPoint.point.geometry.coordinates]
+  const segment = await getSegment(points, true)
+  if (segment && segment.coordinates) clonedPatternSegments[insertionPoint] = segment.coordinates
+}
+
+/**
+ * Converts an object with numerical indices to an array.
+ */
+const objectToArray = (obj): Array<any> => Object.keys(obj)
+  .sort((a, b) => a - b)
+  .reduce((acc, currentKey) => {
+    if (obj[currentKey] === null) return acc
+    return [...acc, obj[currentKey]]
+  }, [])
```

Comment:
Looking at what is going on later in the file, this method seems to just delete null entries from an array.
In that case, you can write:
```suggestion
const removeBlanks = (obj): Array<any> => Object.values(obj).filter(item => item)
```

Reply:
This method wasn't needed anymore after moving back to arrays.